Tests/SentryTests/SentryClientTests.swift

```diff
@@ -1464,24 +1478,48 @@ class SentryClientTest: XCTestCase {
     }
-    
+    @available(*, deprecated, message: "-[SentryClient captureUserFeedback:] is deprecated. -[SentryClient captureFeedback:withScope:] is the new way. This test case can be removed in favor of testNoDsn_FeedbackNotSent when -[SentryClient captureUserFeedback:] is removed.")
     func testNoDsn_UserFeedbackNotSent() {
         let sut = fixture.getSutWithNoDsn()
         sut.capture(userFeedback: UserFeedback(eventId: SentryId()))
         assertNothingSent()
     }
     
+    @available(*, deprecated, message: "-[SentryClient captureUserFeedback:] is deprecated. -[SentryClient captureFeedback:withScope:] is the new way. This test case can be removed in favor of testDisabled_FeedbackNotSent when -[SentryClient captureUserFeedback:] is removed.")
     func testDisabled_UserFeedbackNotSent() {
         let sut = fixture.getSutDisabledSdk()
         sut.capture(userFeedback: UserFeedback(eventId: SentryId()))
         assertNothingSent()
     }
     
+    @available(*, deprecated, message: "-[SentryClient captureUserFeedback:] is deprecated. -[SentryClient captureFeedback:withScope:] is the new way. This test case can be removed in favor of testCaptureFeedback_WithEmptyEventId when -[SentryClient captureUserFeedback:] is removed.")
     func testCaptureUserFeedback_WithEmptyEventId() {
         let sut = fixture.getSut()
         sut.capture(userFeedback: UserFeedback(eventId: SentryId.empty))
         assertNothingSent()
     }
+    
+    @available(*, deprecated, message: "This is only marked as deprecated because assertNothingSent is marked as deprecated, due to it using a deprecated property inside it. When that property usage is removed, this deprecation annotation can go away.")
+    func testNoDsn_FeedbackNotSent() {
+        let sut = fixture.getSutWithNoDsn()
+        sut.capture(feedback: fixture.feedback, scope: fixture.scope)
+        assertNothingSent()
+    }
+    
+    @available(*, deprecated, message: "This is only marked as deprecated because assertNothingSent is marked as deprecated, due to it using a deprecated property inside it. When that property usage is removed, this deprecation annotation can go away.")
+    func testDisabled_FeedbackNotSent() {
+        let sut = fixture.getSutDisabledSdk()
+        sut.capture(feedback: fixture.feedback, scope: fixture.scope)
+        assertNothingSent()
+    }
+    
+    @available(*, deprecated, message: "This is only marked as deprecated because assertNothingSent is marked as deprecated, due to it using a deprecated property inside it. When that property usage is removed, this deprecation annotation can go away.")
+    func testCaptureFeedback_WithEmptyEventId() {
+        let sut = fixture.getSut()
+        XCTAssertTrue(fixture.transportAdapter.sendEventWithTraceStateInvocations.isEmpty)
+        sut.capture(feedback: fixture.feedback, scope: fixture.scope)
+        XCTAssertFalse(fixture.transportAdapter.sendEventWithTraceStateInvocations.isEmpty)
```

Comment:
@philipphofmann LMK what you think of this test. I debugged and traced execution to something I could check, which turned out to be the transport adapter, so just used that. I do see other test cases using `sendEventWithTraceStateInvocations` as well.

Reply:
I think it's OK to do it like that, and the test should assert some properties of the recorded event. Now it only asserts that there was an event recorded but not that the contents of the event are correct.

Reply:
Added checks of the envelope contents in https://github.com/getsentry/sentry-cocoa/pull/4874/commits/530757562689af243b5e11762c59eede7428c509